src/odemis/gui/comp/overlay/world.py

```diff
@@ -1231,46 +1314,350 @@ def draw(self, ctx, selected, shift=(0, 0), scale=1.0, canvas=None):
+            text_left, text_top = self.cnvs.buffer_to_view(self._label.render_pos)
+            text_width, text_height = self._label.text_size
+            text_right = text_left + text_width
+            text_bottom = text_top + text_height
+
+            self._edges.update({
+                "t_l": text_left - self.HOVER_MARGIN,
+                "t_r": text_right + self.HOVER_MARGIN,
+                "t_t": text_top - text_height - self.HOVER_MARGIN,
+                "t_b": text_bottom - text_height + self.HOVER_MARGIN,
+            })
+
+    def stop_updating_tool(self):
+        """ Stop dragging (moving/editing) the label """
+        super(LabelTool, self).stop_updating_tool()
+
+        if self.mode in (CREATE_LABEL, EDIT_TEXT_EDGE_MODE):
+            if not self.motion:
+                self._edit_text()
+            self.calc_edges()
+            self.mode = NONE_LABEL_MODE
+            self.motion = False
+
+    def get_hover(self, vpos):
+        """ Check if the given position is on/near a selection edge or inside the selection.
+        It returns a "gui.HOVER_*" """
+        super(LabelTool, self).get_hover(vpos)
+
+        if self._edges:
+            vx, vy = vpos
+
+            if self._edges:
+                vx, vy = vpos
+
+                if (
+                        self._edges["t_l"] < vx < self._edges["t_r"] and
+                        self._edges["t_t"] < vy < self._edges["t_b"]
+                ):
+                    return gui.HOVER_TEXT
+
+            # if position outside outer box
+            if not (
+                self._edges["o_l"] < vx < self._edges["o_r"] and
+                self._edges["o_t"] < vy < self._edges["o_b"]
+            ):
+                return gui.HOVER_NONE
+
+            # if position inside inner box
+            if (
+                    self._edges["i_l"] < vx < self._edges["i_r"] and
+                    self._edges["i_t"] < vy < self._edges["i_b"]
+            ):
+                dist = util.perpendicular_distance(self.v_start_pos, self.v_end_pos, vpos)
+                if dist < self.HOVER_MARGIN:
+                     return gui.HOVER_LINE
+
+            elif (
+                    self._edges["s_l"] < vx < self._edges["s_r"] and
+                    self._edges["s_t"] < vy < self._edges["s_b"]
+            ):
+                return gui.HOVER_START
+            elif (
+                    self._edges["e_l"] < vx < self._edges["e_r"] and
+                    self._edges["e_t"] < vy < self._edges["e_b"]
+            ):
+                return gui.HOVER_END
+            else:
+                dist = util.perpendicular_distance(self.v_start_pos, self.v_end_pos, vpos)
+                if dist < self.HOVER_MARGIN:
+                    return gui.HOVER_LINE
+
+        return gui.HOVER_NONE
+
+    def debug_edges(self, ctx):
+        super(LabelTool, self).debug_edges(ctx)
+
+        if self._edges and self._label.render_pos:
```

Comment:
Looking for `self._label.render_pos` is a little weird here, as you don't use it. Do you still need that? Or is that to check whether `t_*` are in `_edges`? If so, I'd suggest to explicitly write `'t_l' in self._edges`.